### PR TITLE
feat(timeseries-data-export): add prop to customise timeseries label

### PR DESCRIPTION
**Pull request by mrtraser**

### CODE REVIEW

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-522.gearboxjs.preview.cogniteapp.com

**Comment by polomani** [comment]
we should try to remember to fix typo in commit message (`label`, not `lable`)

File: src/components/TimeseriesDataExport/TimeseriesDataExport.tsx [hunk 1]
```diff
@@ -28,7 +37,10 @@ export type FetchCSVCall = (
   opts: CsvParseOptions
 ) => Promise<string>;
 
-export type FetchTimeseriesCall = (ids: InternalId[]) => Promise<TimeSeries[]>;
+export type FetchTimeseriesCall = (
+  ids: InternalId[]
+) => Promise<GetTimeSeriesMetadataDTO[]>;
+export type LabelFormatterForCSV = LabelFormatter;
```
Comment (polomani) [inline src/components/TimeseriesDataExport/TimeseriesDataExport.tsx]:
```suggestion
export type CSVLabelFormatter = LabelFormatter;
```
maybe?

File: src/components/TimeseriesDataExport/stories/format-label.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+```typescript jsx
+import 'antd/dist/antd.css';
+
+import React from 'react';
+import { GetTimeSeriesMetadataDTO } from '@cognite/sdk'; 
+import { TimeseriesDataExport, LabelFormatterForCSV } from '@cognite/gearbox';
+
+function ExampleComponent(props) {
+  const labelFormatter: LabelFormatterForCSV = 
+    (ts: GetTimeSeriesMetadataDTO) => ts.name || \`timeserie-${ts.id}\`;
```
Comment (polomani) [inline src/components/TimeseriesDataExport/stories/format-label.md]:
```suggestion
    (ts: GetTimeSeriesMetadataDTO) => ts.name; 
```
we know that there is only one timeserie, and it has the name right?

File: src/components/TimeseriesDataExport/TimeseriesDataExport.tsx (hunk 1 above)
Comment (mrtraser) [reply src/components/TimeseriesDataExport/TimeseriesDataExport.tsx]: nope. `CSVLabelFormatter` –  more complex internal interface and here I'm only exporting function type

File: src/components/TimeseriesDataExport/stories/format-label.md (hunk 2 above)
Comment (mrtraser) [reply src/components/TimeseriesDataExport/stories/format-label.md]: yes, but `GetTimeSeriesMetadataDTO ` interface  describes `name` field as optional. So I've placed this default string here

**Comment by mrtraser** [comment]
> we should try to remember to fix typo in commit message (`label`, not `lable`)

done

File: src/components/TimeseriesDataExport/stories/format-label.md (hunk 2 above)
Comment (polomani) [reply src/components/TimeseriesDataExport/stories/format-label.md]: ah, right

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-522.gearboxjs.preview.cogniteapp.com

File: src/components/TimeseriesDataExport/stories/TimeseriesDataExport.stories.tsx [hunk 3]
```diff
@@ -10,9 +10,11 @@ import { MockCogniteClient } from '../../../mocks/mockSdk';
 import { getGranularityInMS } from '../../../utils/utils';
 import { ClientSDKProvider } from '../../ClientSDKProvider';
 import {
+  LabelFormatterForCSV,
```
Comment (polomani) [inline src/components/TimeseriesDataExport/stories/TimeseriesDataExport.stories.tsx]: how this works?

Comment (mrtraser) [reply src/components/TimeseriesDataExport/stories/TimeseriesDataExport.stories.tsx]: fixed

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-522.gearboxjs.preview.cogniteapp.com

**Pull request merged**